velox/exec/Spiller.cpp

```diff
@@ -275,7 +281,7 @@ std::unique_ptr<Spiller::SpillStatus> Spiller::writeSpill(int32_t partition) {
       totalBytes += extractSpillVector(
           run.rows, kTargetBatchRows, kTargetBatchBytes, spillVector, written);
       state_.appendToPartition(partition, spillVector);
-      if (totalBytes > state_.targetFileSize()) {
+      if (state_.compressedSize(totalBytes) > state_.targetFileSize()) {
```

Comment:
Compute the compressedSize every kTargetBatchRows(64) rows, it maybe high expense. We can also estimate the sizeBeforeCompress by state_.targetFileSize(), but it will be complicated, we should use binary search to get the approximate value.

Reply:
Can we just use the actual written file size for this check?

Reply:
No, I think the targetFileSize is equal to the size after data compression.

What do you mean `actual written file size`?

Reply:
You can leverage spilledBytes() from SpillFileList to detect if we have write enough data? Or let SpillFileList::flush() to return the actually written bytes? Thanks!